common/courses/notifications.ts

```diff
@@ -185,8 +185,47 @@ export const canPromoteSubcourse = async (subcourse: Prisma.subcourse, attempted
 export async function sendPupilCoursePromotion(subcourse: Prisma.subcourse, promotionType: Prisma.subcourse_promotion_type_enum) {
     const { allowed, reason } = await canPromoteSubcourse(subcourse, promotionType);
     if (!allowed) {
-        logger.info(`Can't promote Subcourse(${subcourse.id}). Reason: ${reason}`);
-        throw new Error(`Promotion for Subcourse(${subcourse.id}) is not valid!`);
+        if (reason == 'invalid-promotion-type') {
+            logger.info(`Can't promote Subcourse(${subcourse.id}). Reason: ${reason}`);
+            throw new Error(`Promotion type for Subcourse(${subcourse.id}) is not valid!`);
```

Comment:
What is the purpose of this change? Throwing an error will be reported as an internal error to the client.

Reply:
This PR belongs to the Issue: corona-school/project-user#1197

I searched the code and found this way of throwing Errors. How should I throw the Error instead?

Reply:
Extend https://github.com/corona-school/backend/blob/master/common/util/error.ts with a new error class, maybe "NotAllowedError"

Reply:
That error class should have the reason as property

Reply:
Then adapt formatMessage in https://github.com/corona-school/backend/blob/master/graphql/error.ts to pass that reason to the client. It can then be catched and handled in the frontend.

Reply:
Maybe also adapt https://github.com/corona-school/backend/blob/master/common/util/decision.ts assertAllowed to throw that new error class and reuse that here

Reply:
I created a new commit to implement your suggestion. However this seems like duplicate code to me. Maybe it would make more sense to export a new client error variable called "NotAllowedError" and then give the reason as the publicMessage?

Reply:
Ja well "That error class should have the reason as property" then it would not be redundant. 

But would also be OK to just create a new error type and then set the message to the reason

Reply:
throw new NotAllowedError(reason); ?

Reply:
Yes I looked at how the other types of client errors were thrown and imitated that.

Reply:
Again: throw new NotAllowedError(reason); - there is no point in showing english errors to our users, translation to a text should be done in the frontend.